src/__tests__/resolver.spec.ts

```diff
@@ -1353,18 +1359,33 @@ describe('resolver', () => {
           bar: {
             hello: 'world',
           },
+          test: {
+            $ref: 'custom://missing-test',
+          },
         },
       });
-      expect({ ...result.errors[0], uri: undefined }).toEqual({
-        code: 'RESOLVE_URI',
-        message: 'Error: not found!',
-        uriStack: [],
-        pointerStack: [],
-        path: ['definitions', 'foo'],
-        uri: undefined,
-      });
-      expect(result.errors[0].uri.toString()).toBe('custom://missing/');
-      expect(result.errors.length).toEqual(1);
+      expect(result.errors).toEqual([
+        {
+          code: 'RESOLVE_URI',
+          message: 'Error: not found!',
+          uriStack: [],
+          pointerStack: [],
+          path: ['definitions', 'foo'],
+          uri: expect.objectContaining({
+            _string: 'custom://missing/',
+          }),
+        },
+        {
+          code: 'RESOLVE_URI',
+          message: 'Error: not found!',
+          uriStack: ['custom://test/'],
+          pointerStack: [],
+          path: [],
```

Comment:
Hmm, is this correct?

Reply:
I think so, since we try resolving the following object
```
 {
     $ref: 'custom://missing-test',
 },
``` 
`$ref` is a top-level property here.

If I nest it, I do get an error
![image](https://user-images.githubusercontent.com/9273484/71384665-ea2ee580-25e2-11ea-8660-8824cac136d1.png)